src/surge-xt/osc/OpenSoundControl.cpp

```diff
@@ -267,5 +297,91 @@ void OSCListener::oscBundleReceived(const juce::OSCBundle &bundle)
     }
 }
 
+/* ----- OSC Sending  ----- */
+
+bool OpenSoundControl::initOSCOut(int port)
+{
+    // Send OSC messages to localhost:UDP port number:
+    if (!juceOSCSender.connect("127.0.0.1", port))
+    {
+#ifdef DEBUG
+        std::cout << "Surge OSCSender: failed to connect to UDP port " << port << "." << std::endl;
+#endif
+        return false;
+    }
+    sendingOSC = true;
+    oportnum = port;
+    synth->storage.oscSending = true;
+
+#ifdef DEBUG
+    std::cout << "SurgeOSC: Sending OSC on port " << port << "." << std::endl;
+#endif
+    return true;
+}
+
+void OpenSoundControl::stopSending()
+{
+    if (!sendingOSC)
+        return;
+
+    sendingOSC = false;
+    synth->storage.oscSending = false;
+#ifdef DEBUG
+    std::cout << "SurgeOSC: Stopped sending OSC." << std::endl;
+#endif
+}
+
+void OpenSoundControl::send(std::string addr, std::string msg)
+{
+    if (sendingOSC)
+    {
+        // Runs on the juce messenger thread
+        juce::MessageManager::getInstance()->callAsync([this, msg, addr]() {
+            if (!this->juceOSCSender.send(juce::OSCMessage(juce::String(addr), juce::String(msg))))
+                std::cout << "Error: could not send OSC message.";
+        });
+    }
+}
+
+// Loop through all params, send them to OSC Out
+void OpenSoundControl::sendAllParams()
+{
+    if (sendingOSC)
+    {
+        // Runs on the juce messenger thread
+        juce::MessageManager::getInstance()->callAsync([this]() {
+            // auto timer = new Surge::Debug::TimeBlock("ParameterDump");
+            std::string valStr;
+            int n = synth->storage.getPatch().param_ptr.size();
+            for (int i = 0; i < n; i++)
+            {
+                Parameter p = *synth->storage.getPatch().param_ptr[i];
+                switch (p.valtype)
+                {
+                case vt_int:
+                    valStr = std::to_string(p.val.i);
+                    break;
+
+                case vt_bool:
+                    valStr = std::to_string(p.val.b);
+                    break;
+
+                case vt_float:
+                    valStr = std::to_string(p.val.f);
```

Comment:
Out of curisoutiy do we need to worry about locales here? `std::to_string` will make pi `3,1415` in france and stuff

Reply:
hadn't considered any i8n issues on this; I think it does need consideration now that you point it out.

Reply:
```
inline std::string float_to_clocalestr(float value)
{
    return fmt::format(std::locale::classic(), "{:L}", value);
}
```

there's something which puts into the `C` locale for you from a float but lets do that separately I agree

Reply:
So I'll merge this one and then we can think about that separately.